app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/VpnTrackerNotificationUpdates.kt

```diff
@@ -54,6 +55,11 @@ class VpnTrackerNotificationUpdates @Inject constructor(
         }
     }
 
+    override fun onVpnReconfigured(coroutineScope: CoroutineScope) {
+        logcat { "Notification updates: onVpnReconfigured" }
+        onVpnStarted(coroutineScope)
```

Comment:
is onVpnReconfigured() necesary if we are doing the  same as onVPNStarted anyway?

Reply:
I think it is. If you don't have a re-configure we'll have to execute **all** onStarted callbacks, which I don't think we want to.
Only those plugins that want/need to do something will override the reconfigured.

Another option is adding a boolean flag to onStarted callbacks. But I have unpopular opinions about boolean flags of that nature, [I think they're an anti-pattern](https://martinfowler.com/bliki/FlagArgument.html).